### ISSUE TITLE
Cannot start docker container from Docker Desktop in Win 10 with WSL2

### ISSUE THREAD

**Issue by justinmraj1984** [docker/for-win#13232]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 3680fe36ac2cff9c8e50e298c2dffbc132470beb2f00d1f825f584eb92edd17b

### Actual behavior
Getting the following error while trying to start the container from Docker Desktop or from the command line

**> docker run -d -p 80:80 docker/getting-started**

> 3680fe36ac2cff9c8e50e298c2dffbc132470beb2f00d1f825f584eb92edd17b
> docker: Error response from daemon: failed to create shim task: OCI runtime create failed: unable to retrieve OCI runtime error (open /var/run/desktop-containerd/daemon/io.containerd.runtime.v2.task/moby/3680fe36ac2cff9c8e50e298c2dffbc132470beb2f00d1f825f584eb92edd17b/log.json: no such file or directory): runc did not terminate successfully: exit status 139: unknown.
> 

### Expected behavior
Executing the above command should start the container - "docker/getting-started"
### Information

  - Windows Version: 10
  - Docker Desktop Version: v4.16.3
  - WSL2 or Hyper-V backend?   WSL2
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: No

### Output of `& "C:\Program Files\Docker\Docker\resources\com.docker.diagnose.exe" check`
[docker_diag.txt](https://github.com/docker/for-win/files/10688058/docker_diag.txt)

### Steps to reproduce the behavior

  1. Install docker desktop v4.16.3
  2. Start the docker desktop after the installation
  3. Under "Containers" section in Docker Desktop, make use of the suggested option to create the container - "docker/getting-started"
  4. The images are are getting pulled successfully, and the container is in "created" status.
  5. On clicking the start icon under actions column, following error occurs
  

> docker: Error response from daemon: failed to create shim task: OCI runtime create failed: unable to retrieve OCI runtime error (open /var/run/desktop-containerd/daemon/io.containerd.runtime.v2.task/moby/3680fe36ac2cff9c8e50e298c2dffbc132470beb2f00d1f825f584eb92edd17b/log.json: no such file or directory): runc did not terminate successfully: exit status 139: unknown.

  6. Same error occurs while running the docker run command to start this container from windows powershell or command line tool.

**Comment by YouveGotMeowxy** [comment]
Didn't replicate for me:

```
docker run -d -p 80:80 docker/getting-started
Unable to find image 'docker/getting-started:latest' locally
latest: Pulling from docker/getting-started
c158987b0551: Already exists
1e35f6679fab: Pull complete
cb9626c74200: Pull complete
b6334b6ace34: Pull complete
f1d1c9928c82: Pull complete
9b6f639ec6ea: Pull complete
ee68d3549ec8: Pull complete
33e0cbbb4673: Pull complete
4f7e34c2de10: Pull complete
Digest: sha256:d79336f4812b6547a53e735480dde67f8f8f7071b414fbd9297609ffb989abc1
Status: Downloaded newer image for docker/getting-started:latest
439e779a5256d4600ab9faa0db406de5cfddbf0f2e8a6e657544c2e93db57066
docker: Error response from daemon: driver failed programming external connectivity on endpoint keen_feistel (d00242c516c18a7ef8ef2058efe37309aec9103dd7851e0edb3d04df9eb17462): listen tcp4 0.0.0.0:80: bind: address already in use.

docker run -d -p 8081:80 docker/getting-started
e99c8a7d4b4e5c16613829efaed6ed4b457b6835f7cff1660580f0c7c2773607
```

I got an error at first only because my port 80 was in use, but changing that port and running it again gave no error.

**Comment by Marineepo** [comment]
go here ---> https://docs.docker.com/desktop/release-notes/
you need to uninstall everything that pertains to docker on your system, then restart your system
once you go to that link,  you need to downgrade install to 4.11.1 I believe which will replace all the newer updates regarding 4.16.3.....

**Comment by aros3rg3d** [comment]
Hi ! 
I Just ran into this issue after upgrading to 4.17.0. I uninstalled it and reinstalled it and now it works.

**Comment by justinmraj1984** [comment]
Hi All !
This issue is resolved for me now. Hence taking this to closure.
Root cause:
1. Installed Docker to use WSL2 in Windows 10
2. By default the WSL2 linux image was created under C drive
3. I took a backup of this image and moved to D drive in my laptop
4. Unregistered the WSL2 docker (based out of C drive), and imported the image from D drive
5. Docker container didn't start after importing the image from D drive - this is mostly an OS level permission issue.

Resolution:
Changed the image location back to C drive, and it us working fine now.

Thanks to all for the suggestions to resolve this issue, Appreciate it !
